### PR TITLE
Implement support for PCI-based VPU

### PR DESCRIPTION
**Pull request by chunfungintel**

### CODE REVIEW

File: cmd/vpu_plugin/vpu_plugin_test.go [hunk 1]
```diff
@@ -62,6 +103,69 @@ func (n *fakeNotifier) Notify(newDeviceTree dpapi.DeviceTree) {
...
+func TestScanPci(t *testing.T) {
+	var fN fakeNotifier
+
+	f, err := os.Create(hddlSocketPci)
+	if err != nil {
+		t.Error("create fake hddl file failed")
+	}
+
+	//create a temporary folder to create fake devices files for PCI scanning
+	tmpPciDir, err := os.MkdirTemp("/tmp", "fake-pci-devices")
```
Comment (mythi) [inline cmd/vpu_plugin/vpu_plugin_test.go:115]: this is available only in Go v1.16 which we don't have yet so the CI test fails.

Comment (chunfungintel) [reply cmd/vpu_plugin/vpu_plugin_test.go:115]: Any suggestion what we should update?

Comment (mythi) [reply cmd/vpu_plugin/vpu_plugin_test.go:115]: #630 has examples for the "inverse":  `ioutil.TempDir`

Comment (chunfungintel) [reply cmd/vpu_plugin/vpu_plugin_test.go:115]: Shall I wait for your update be merged first?

Comment (mythi) [reply cmd/vpu_plugin/vpu_plugin_test.go:115]: let's not wait that but use `ioutil.TempDir` in this PR.

**Comment by mythi** [comment]
@chunfungintel this looks OK to me but it seems I need to fix the image builds first.

@rojkov @bart0sh any final code review comments?

**Comment by bart0sh** [comment]
@chunfungintel I'd suggest to squash commits. Otherwise lgtm.

**Review by rojkov — DISMISSED** [review]
LGTM. Any refinements can go to follow-up PRs.

**Comment by mythi** [comment]
@chunfungintel can you rebase to latest main and squash the commits.

**Review by mythi — APPROVED** [review]

**Pull request merged**